diff --git a/modules/contrib/group/src/Access/GroupPermissionHandler.php b/modules/contrib/group/src/Access/GroupPermissionHandler.php
index 54a839ab..8a723e36 100644
--- a/modules/contrib/group/src/Access/GroupPermissionHandler.php
+++ b/modules/contrib/group/src/Access/GroupPermissionHandler.php
@@ -281,15 +281,17 @@ protected function sortPermissions(array $permissions = []) {
     // Sort all permissions by provider, section and title, in that order.
     uasort($permissions, function (array $permission_a, array $permission_b) use ($modules) {
       if ($permission_a['provider'] == $permission_b['provider']) {
-        if ($permission_a['section'] == $permission_b['section']) {
-          // All permissions should have gone through ::completePermission() so
-          // the titles are \Drupal\Core\StringTranslation\TranslatableMarkup.
-          $title_a = $permission_a['title']->__toString();
-          $title_b = $permission_b['title']->__toString();
-          return strip_tags($title_a) > strip_tags($title_b);
-        }
-        else {
-          return $permission_a['section'] > $permission_b['section'];
+        if (isset($permission_a['section']) && isset($permission_b['section'])) {
+          if ( $permission_a['section'] == $permission_b['section']) {
+            // All permissions should have gone through ::completePermission() so
+            // the titles are \Drupal\Core\StringTranslation\TranslatableMarkup.
+            $title_a = $permission_a['title']->__toString();
+            $title_b = $permission_b['title']->__toString();
+            return strip_tags($title_a) > strip_tags($title_b);
+          }
+          else {
+            return $permission_a['section'] > $permission_b['section'];
+          }
         }
       }
       else {
diff --git a/modules/contrib/group/src/Form/GroupPermissionsForm.php b/modules/contrib/group/src/Form/GroupPermissionsForm.php
index c8e757b5..00e7bfe7 100644
--- a/modules/contrib/group/src/Form/GroupPermissionsForm.php
+++ b/modules/contrib/group/src/Form/GroupPermissionsForm.php
@@ -103,7 +103,9 @@ protected function getPermissions() {
 
     // Create a list of group permissions ordered by their provider and section.
     foreach ($this->groupPermissionHandler->getPermissionsByGroupType($this->getGroupType()) as $permission_name => $permission) {
-      $by_provider_and_section[$permission['provider']][$permission['section']][$permission_name] = $permission;
+      if (isset($permission['section'])) {
+        $by_provider_and_section[$permission['provider']][$permission['section']][$permission_name] = $permission;
+      }
     }
 
     // Always put the 'General' section at the top if provided.
